### PR TITLE
Bump Nessie to version 0.4.0

### PR DESCRIPTION
**Pull request by snazy**
Contains some "noise" due to the renamed package `com.dremio.nessie` -> `org.projectnessie`.

Implicitly bumps Quarkus to 1.12.1.Final

Quarkus image building requires us to set the system property `quarkus.build.native-image`, which is not particularly
great and should be changed in the future on the Nessie-Quarkus-Gradle-Plugin side. It is a change in the behavior
since Quarkus 1.10 (Nessie 0.3.0 used Quarkus 1.9.1.Final) regarding how the Quarkus platform properties are applied,
which sadly also affects Quarkus builds that do not build a native image. Playing around with applying the Quarkus
bom, as suggested by Quarkus project, in various places (Gradle configs) didn't help here. I suspect, that's some
transitive dependency resolving issue.

### CODE REVIEW

File: nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ NessieCatalog initCatalog(String ref) {
     NessieCatalog newCatalog = new NessieCatalog();
     newCatalog.setConf(hadoopConfig);
     newCatalog.initialize("nessie", ImmutableMap.of("ref", ref,
-        "url", path,
+        "uri", path,
```
Comment (rymurr) [inline nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java:106]: this can use `CatalogProperties`

File: build.gradle [hunk 2]
```diff
@@ -1102,7 +1102,10 @@ project(':iceberg-nessie') {
   quarkusAppRunnerProperties {
     props = ["quarkus.http.test-port": 0]
   }
-  tasks.getByName("quarkus-start").dependsOn("compileTestJava")
+  tasks.getByName("quarkus-start")
+          // This is not doing anything with Docker or building a native image, just a quirk of in Quarkus since 1.10.
```
Comment (rymurr) [inline build.gradle:1106]: > quirk of in Quarkus 

nit: quirk of Quarkus

**Review by rymurr — COMMENTED** [review]
Thanks @snazy looks great! Just a few minor nits

**Review by rymurr — APPROVED** [review]
LGTM

File: build.gradle (hunk 2 above)
Comment (RussellSpitzer) [reply build.gradle]: quark of quirkus

**Comment by danielcweeks** [comment]
+1

**Comment by rymurr** [comment]
Much obliged @danielcweeks!

**Pull request merged**